frontend/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -1,35 +1,92 @@
-import React, { FunctionComponent, ReactElement } from "react";
+import React, {
+  FunctionComponent,
+  ReactElement,
+  useEffect,
+} from "react";
 import { connect } from "react-redux";
 import { useHistory } from "react-router-dom";
 import styled from "styled-components";
 import Layout from "../../components/Layout";
-import { SignOut } from "./data/actions";
-import { getIsSignedOut } from "./data/reducer";
+import PurpleButton from "../../components/PurpleButton";
+import { User } from "../../models/types";
+import {
+  GetData,
+  GetDataRequestProps,
+  SignOut,
+} from "./data/actions";
+import { getCurrentUser, getIsSignedOut } from "./data/reducer";
 
 const SignOutButton = styled.button``;
 
 interface DashboardParams {
-  isSignedOut: boolean,
-  signOut: () => void
+  isSignedOut: boolean;
+  currentUser?: User;
+  signOut: () => void;
+  getUser: (val: GetDataRequestProps) => void;
 }
 
-const connector = connect(state => ({
-  isSignedOut: getIsSignedOut(state)
-}), {
-  signOut: SignOut.request
-})
+const connector = connect(
+  (state) => ({
+    isSignedOut: getIsSignedOut(state),
+    currentUser: getCurrentUser(state),
+  }),
+  {
+    signOut: SignOut.request,
+    getUser: GetData.request,
+  }
+);
 
-const Dashboard : FunctionComponent<DashboardParams> = ({ isSignedOut, signOut }) : ReactElement => {
+const getTitleOfButton = (user: User): string => {
+  switch (user.sessionId) {
+    case -1:
+      return "Begin pre-assessment";
+    case 9:
+      return "Congratulations on finishing the study";
+    default:
+      return (
+        (user.onAssessment ? "Continue session " : "Begin session ") +
+        (user.sessionId + 1)
```

Comment:
we probably want to stop if sessionID is 8 right or some number

Reply:
It does stop (see the `disabled` flag)